### ISSUE TITLE
russian: prepositions are too long

### ISSUE THREAD

**Issue by beqabeqa473** [espeak-ng/espeak-ng#544]
hello all.

in russian language implementations, prepositions such as с в к and so on are too long.

they should be shortened a bit
example phrases

мальчик пошел в школу
кирпич упал с крышы
собака прибежала к тебе

please fix these issues

**Comment by valdisvi** [comment]
This was already done in commit 6cc1c009007d78900852fec1d84cd371a45526dc and then issue 
#474 was reported, that single letter words are too short.

**Comment by beqabeqa473** [comment]
hi.

yes, for that time, single letter words and spelled letters sounded the same.
in russian, there should be a difference
for example, take letter в
when spelling letter в, it should sound вэб but as preposition it
should sound without э, i mean it should be short.

what could be done to improve the situation?

thanks.

2018-09-06 18:57 GMT+04:00, Valdis Vitolins <notifications@github.com>:
> This was already done in commit 6cc1c009007d78900852fec1d84cd371a45526dc and
> then issue #474 was reported, that single letter words are too short.
>
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/espeak-ng/espeak-ng/issues/544#issuecomment-419125253


-- 
with best regards beqa

**Comment by Qchristensen** [comment]
I also have reports from eSpeak-NG users of NVDA with the same issues.  One stated:

eSpeak-NG put to much emphasis on prepositions, for example (в с на).

Some words are also misread. example (интернет.)

**Comment by beqabeqa473** [comment]
hi.

yes, i can confirm every issue stated by users.

massive problems started to happen after valdis's changes.

2018-09-26 13:06 GMT+04:00, Quentin Christensen <notifications@github.com>:
> I also have reports from eSpeak-NG users of NVDA with the same issues.  One
> stated:

>

> eSpeak-NG put to much emphasis on prepositions, for example (в с на).

>

> Some words are also misread. example (интернет.)

>
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/espeak-ng/espeak-ng/issues/544#issuecomment-424642173


-- 
with best regards beqa

**Comment by rhdunn** [comment]
I checked wiktionary when making the `в` and `с` changes, but either I misread it or it has since changed. As such, I have reverted those two prepositions to their eSpeak pronunciations.

Regarding the pronunciation of `на`, I have modified it to an unstressed `n@` when used as a preposition. When it appears at the end (e.g. as just that word), it uses `na` to reflect the version meaning "here!" per wiktionary.

Regarding `интернет` (internet), wiktionary lists a phonetic respelling (`интэрнэ́т`) which eSpeak NG pronounces like the English word. My reading of that is that the pronunciation of `интернет` is correct (`т` and`н` before `е` are palatalized). However, I have added an exception for this word due to the intended usage.

@beqabeqa473 The previous Russian changes have been reverted to the old eSpeak versions. The subsequent changes have mainly been made by me, not Valdis, so don't blame him. Also, Valdis and I are doing our best to improve and maintain eSpeak (which would have been abandoned, and not have emoji support among other things).

The changes from after the reset to the eSpeak pronunciations are:
1.  `н` (`n`) is softened (palatalized -- a `y` sound) before iotated vowel (`ь` `ю` `я` `ё`)
1.  Use `;` for palatalization instead of `^` in palatalized `n` and `l`
1.  The particle `б` was shortened to `бы` (https://en.wiktionary.org/wiki/%D0%B1#Russian)
1.  The prepositions `в` and `с` were changed to add an `/ɵ/` sound.
1.  A Russian (Latvia) accent was added.
1.  The changes described above.

**Comment by beqabeqa473** [comment]
hi.

yes, i understand that you are doing the best to improve things, for
that we are very thankfull.

why i am talking as i am talking about valdis is that he were trying
to say that those changes are similar to russian grammar rules.
however i said that it is not correct, after that he gave me this
stupid recording again mostly with non native speakers.
i can confidently say, that he doesn't know language on such level to
make so big decisions to change things.

i think you understood why i was talking so negatively about him.

again, i don't have any problem about him, but he must realise that if
you want to make some evolution in language support, you should think
and listen to firstly about users who are using espeak in russian, and
not some latvian people who are on the same level as he is in terms of
language knowledge and only after that make them.

2018-09-27 1:08 GMT+04:00, erikpro <notifications@github.com>:
> @Qchristensen hello. wood it be possible to add that version in nvda
> master?
> or maybe someone can teach me how to compile espeak-ng as an addon?
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/espeak-ng/espeak-ng/issues/544#issuecomment-424870380


-- 
with best regards beqa

**Comment by beqabeqa473** [comment]
as for интернет

i cannot reproduce a problem.
i understand that there is somehow problem with не sometimes in the
middle of the word, but again i cannot hear this problem

2018-09-27 1:08 GMT+04:00, erikpro <notifications@github.com>:
> @Qchristensen hello. wood it be possible to add that version in nvda
> master?
> or maybe someone can teach me how to compile espeak-ng as an addon?
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/espeak-ng/espeak-ng/issues/544#issuecomment-424870380


-- 
with best regards beqa
